src/sql/src/query_model/hir_generator.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
```

Comment:
Likewise, add some module doc that just says something like "Generates a graph query model from a [HirRelationExpr]"